proto/ai/traceable/agent/config/v1/config.proto

```diff
@@ -182,3 +241,120 @@ message RateLimitConfig {
       // set the span type for rate limited spans
       SpanType span_type = 6;
 }
+
+// Message describes what message should be considered for certain DataCapture option
+message Message {
+  // when `false` it disables the capture for the request in a client/request operation
+  google.protobuf.BoolValue request = 1;
+
+  // when `false` it disables the capture for the response in a client/request operation
+  google.protobuf.BoolValue response = 2;
+}
+
+// DataCapture describes the elements to be captured by the agent instrumentation
+message DataCapture {
+  // http_headers enables/disables the capture of the request/response headers in HTTP
+  Message http_headers = 1;
+
+  // http_body enables/disables the capture of the request/response body in HTTP
+  Message http_body = 2;
+
+  // rpc_metadata enables/disables the capture of the request/response metadata in RPC
+  Message rpc_metadata = 3;
+
+  // rpc_body enables/disables the capture of the request/response body in RPC
+  Message rpc_body = 4;
+
+  // body_max_size_bytes is the maximum size of captured body in bytes.
+  // Default should be 131_072 (128 KiB).
+  google.protobuf.Int32Value body_max_size_bytes = 5;
+
+  // body_max_processing_size_bytes is maximum size of body being processed by filters in bytes.
+  // Default should be 1_048_576 (1MB).
+  //
+  // For uncompressed bodies we capture all bytes up to `body_max_processing_size_bytes`
+  // in memory and pass that through the filter.
+  // For compressed and GRPC bodies, if the size of the body is larger than this, we ignore
+  // it entirely, otherwise we decompress/decode the body and then pass it to the filter.
+  google.protobuf.Int32Value body_max_processing_size_bytes = 6;
+
+  // Array of allowed content type substrings to record
+  // default should be json, x-www-form-urlencoded
+  // ex: ["json"] will record any request bodies that have a content-type header that includes "json"
+  repeated google.protobuf.StringValue allowed_content_types = 10;
+}
+
+// PropagationFormat represents the propagation formats supported by agents
+enum PropagationFormat {
+  // B3 propagation format, agents should support both multi and single value formats
+  // see https://github.com/openzipkin/b3-propagation
+  B3 = 0;
+
+  // W3C Propagation format
+  // see https://www.w3.org/TR/trace-context/
+  TRACECONTEXT = 1;
+}
+
+// TraceReporterType represents the reporting format for trace data.
+enum TraceReporterType {
+
+  // Default to none. Agent will use it's default reporting type
+  UNSPECIFIED = 0;
+
+  // Zipkin protobuf reporting format.
+  // see https://github.com/openzipkin/zipkin-api
+  ZIPKIN = 1;
+
+  // OpenTelemetry protobuf reporting format.
+  // see https://github.com/open-telemetry/opentelemetry-proto
+  OTLP = 2;
+
+  // Logging reporting format
+  LOGGING = 3;
+
+  // Disable trace reporting
+  NONE = 4;
+
+  // OTLP over http
+  OTLP_HTTP = 5;
```

Comment:
We can avoid buf linting these in the buf.yml file. I think the config is like this
```
ignore_only:
    ENUM_ZERO_VALUE_SUFFIX:
      - config/common/v1/config.proto
    ENUM_VALUE_PREFIX:
      - config/common/v1/config.proto
```

Reply:
I tried that(a few iterations of it) but that doesn't appear to work for at least our version(I see it referenced in the v1 and v2 docs but unsure of v1beta1) I used except instead which I think these configs would be analogous unless we add a new file.